lib/solarwinds_apm/sampling/oboe_sampler.rb

```diff
@@ -293,41 +285,42 @@ def disabled_algo(sample_state)
         @logger.debug { 'SAMPLE_THROUGH_ALWAYS is set; record' }
         sample_state.decision = OTEL_SAMPLING_DECISION::RECORD_ONLY
       end
+      @logger.debug { "[#{self.class}/#{__method__}] disabled_algo end" }
     end
 
     def update_settings(settings)
       return unless settings[:timestamp] > (@settings[:timestamp] || 0)
 
-      @settings = settings
-      @buckets.each do |type, bucket|
-        bucket.update(@settings[:buckets][type]) if @settings[:buckets][type]
+      @settings_mutex.synchronize do
```

Comment:
Going to be extra picky: please could you update the PR description and/or add a comment about this? I'm assuming it's for thread safety.

Reply:
Sure